cli/src/main.rs

```diff
@@ -141,18 +148,48 @@ fn main_inner() -> Result<(), String> {
             let formula_path = solver_args.value_of("formula").unwrap();
             let formula_format = get_formula_format_from_args(&solver_args)?;
             let search_strategy = get_search_strategy_from_args(&solver_args)?;
+            let game_strategy_path = solver_args.value_of("game_strategy");
 
             // Generic start function for use with `load` that start model checking with `distributed_certain_zero`
             fn check_model<G>(
                 graph: ATLDependencyGraph<G>,
                 v0: ATLVertex,
                 threads: u64,
                 ss: SearchStrategyOption,
-            ) where
+                game_strategy_path: Option<&str>,
+            ) -> Result<(), String>
+            where
                 G: GameStructure + Send + Sync + Clone + Debug + 'static,
             {
-                let result = ss.distributed_certain_zero(graph, v0, threads);
-                println!("Result: {}", result);
+                let result =
+                    ss.model_check(graph.clone(), v0, threads, game_strategy_path.is_some());
+                println!("Result: {}", &result.satisfied);
+
+                if let Some(game_strategy_path) = game_strategy_path {
+                    let proof_res = result.proof.unwrap();
+                    match proof_res {
+                        Ok(proof) => match proof {
+                            SpecificationProof::Strategy(strategy) => {
+                                let mut file = File::create(game_strategy_path).map_err(|err| {
+                                    format!("Failed to create strategy output file.\n{}", err)
+                                })?;
+                                write!(file, "{}", strategy.in_context_of(&graph.game_structure))
+                                    .map_err(|err| {
+                                        format!("Failed to write strategy file. {}", err)
+                                    })?;
+                                println!("Proving strategy was save to {}", game_strategy_path);
+                            }
+                            SpecificationProof::NoStrategyNeeded => {
+                                println!("No game strategy was computed since a strategy is not needed to prove the given query.")
+                            }
+                        },
+                        Err(err) => {
+                            println!("Game strategy was not computed due to error: {}", err)
+                        }
+                    }
+                }
```

Comment:
At some point, this stdout flow needs to match current main on handling quiet input